File: LeetCode/259-3sum-smaller/ex_259.py
class Solution(object):
    def threeSumSmaller(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        count = 0
        nums.sort()
        for i in xrange(len(nums) - 2):
            t = target - nums[i]
            j = i + 1
            k = len(nums) - 1
            while j < k:
                if nums[j] + nums[k] < t:   
                    count += k - j    
                    j += 1
                else:
                    k -= 1
        return count

File: LeetCode/259-3sum-smaller/ex_259.cpp
class Solution {
public:
    int threeSumSmaller(vector<int>& nums, int target) {
        if (nums.size() < 3)  return 0;
        int count = 0;
        sort(nums.begin(), nums.end());
        for (int i = 0; i < nums.size() - 2; ++i) {
            int tar = target - nums[i];
            int j = i + 1;
            int k = nums.size() - 1;
            while (j < k) {
                if (nums[j] + nums[k] < tar) {
                    count += k - j;
                    ++j;
                }
                else    --k;
            }
        }
        return count;
    }
};